File: cash/cash.c
#include <stdio.h>
#include <stdlib.h>

int main(void)
{
    // Variabel for å holde mengden som skal endres
    int changeOwed;
    // Variabel for å holde brukerens inndata
    char userInput[100];
    // Variabel for å sjekke om inndata er gyldig
    int isValidInput = 0;

    // Spør brukeren om mengden endring tilknyttet og valider inndata
    while (!isValidInput)
    {
        printf("Endring skyldig: ");
        fgets(userInput, sizeof(userInput), stdin);

        // Sjekk om inndata er et gyldig heltall
        if (sscanf(userInput, "%d", &changeOwed) == 1 && changeOwed > 0)
        {
            isValidInput = 1;
        }
        else
        {
            printf("Ugyldig inndata. Vennligst skriv inn et positivt heltall.\n");
        }
    }

    // Variabler for antall mynter
    int numberOfQuarters = 0;
    int numberOfDimes = 0;
    int numberOfNickels = 0;
    int numberOfPennies = 0;

    // Beregn antall kvartaler (25¢)
    while (changeOwed >= 25)
    {
        changeOwed -= 25;
        numberOfQuarters++;
    }

    // Beregn antall ti-cent mynter (10¢)
    while (changeOwed >= 10)
    {
        changeOwed -= 10;
        numberOfDimes++;
    }

    // Beregn antall fem-cent mynter (5¢)
    while (changeOwed >= 5)
    {
        changeOwed -= 5;
        numberOfNickels++;
    }

    // Beregn antall en-cent mynter (1¢)
    while (changeOwed >= 1)
    {
        changeOwed -= 1;
        numberOfPennies++;
    }

    // Beregn totalt antall mynter
    int totalCoins = numberOfQuarters + numberOfDimes + numberOfNickels + numberOfPennies;

    // Skriv ut resultatet
    printf("%d\n", totalCoins);

    return 0;
}

File: sentimental-cash/cash.py
# Variabel for å holde mengden som skal endres
change_owed = 0
is_valid_input = False

# Spør brukeren om mengden endring tilknyttet og valider inndata
while not is_valid_input:
    try:
        # Forsøker å konvertere til et flyttall
        change_owed = float(input("Change owed: "))
        if change_owed > 0:
            is_valid_input = True  # Godtar innput dersom det er et positivt tall
        else:
            print("Invalid input. Please enter a positive number.")
    except ValueError:
        # Håndterer ikke-numerisk input
        print("Invalid input. Please enter a positive number.")

# Konverter til hele cent
change_owed = round(change_owed * 100)

# Variabler for antall mynter
number_of_quarters = 0
number_of_dimes = 0
number_of_nickels = 0
number_of_pennies = 0

# Beregn antall kvartaler (25¢)
while change_owed >= 25:
    change_owed -= 25
    number_of_quarters += 1

# Beregn antall deser (10¢)
while change_owed >= 10:
    change_owed -= 10
    number_of_dimes += 1

# Beregn antall nickler (5¢)
while change_owed >= 5:
    change_owed -= 5
    number_of_nickels += 1

# Beregn antall pennere (1¢)
while change_owed >= 1:
    change_owed -= 1
    number_of_pennies += 1

# Beregn totalt antall mynter
total_number_of_coins = number_of_quarters + \
    number_of_dimes + number_of_nickels + number_of_pennies

print(f"{total_number_of_coins}")
